testsuite/testcases/src/multi_region_network_test.rs

```diff
@@ -0,0 +1,314 @@
+    );
+
+    let validator_chunks = chunk_validators(validators, number_of_regions);
+
+    let validator_groups = validator_chunks
+        .into_iter()
+        .zip(link_stats_table.iter())
+        .map(|(chunk, (from_region, stats))| (from_region.clone(), chunk, stats.clone()))
+        .collect();
+
+    validator_groups
+}
+
+// A map of "source" regions to a map of "destination" region to (bandwidth, latency)
+type LinkStatsTable = BTreeMap<String, BTreeMap<String, (u64, f64)>>;
+// A map of "source" regions to a tuple of (list of validators, map of "destination" region to (bandwidth, latency))
+type LinkStatsTableWithPeerGroups = Vec<(String, Vec<PeerId>, BTreeMap<String, (u64, f64)>)>;
+
+#[derive(Clone)]
+pub struct InterRegionNetEmConfig {
+    delay_jitter_ms: u64,
+    delay_correlation_percentage: u64,
+    loss_percentage: u64,
+    loss_correlation_percentage: u64,
+}
+
+impl Default for InterRegionNetEmConfig {
+    fn default() -> Self {
+        Self {
+            delay_jitter_ms: 20,
+            delay_correlation_percentage: 50,
+            loss_percentage: 3,
+            loss_correlation_percentage: 50,
+        }
+    }
+}
+
+impl InterRegionNetEmConfig {
+    // Creates GroupNetEm for inter-region network chaos
+    fn build(&self, validator_groups: &LinkStatsTableWithPeerGroups) -> Vec<GroupNetEm> {
+        let group_netems: Vec<GroupNetEm> = validator_groups
+            .iter()
+            .combinations(2)
+            .map(|comb| {
+                let (from_region, from_chunk, stats) = &comb[0];
+                let (to_region, to_chunk, _) = &comb[1];
+
+                let (bandwidth, latency) = stats.get(to_region).unwrap();
+                let netem = GroupNetEm {
+                    name: format!("{}-to-{}-netem", from_region, to_region),
+                    source_nodes: from_chunk.to_vec(),
+                    target_nodes: to_chunk.to_vec(),
+                    delay_latency_ms: *latency as u64,
+                    delay_jitter_ms: self.delay_jitter_ms,
+                    delay_correlation_percentage: self.delay_correlation_percentage,
+                    loss_percentage: self.loss_percentage,
+                    loss_correlation_percentage: self.loss_correlation_percentage,
+                    rate_in_mbps: *bandwidth / 1e6 as u64,
+                };
+                info!("inter-region netem {:?}", netem);
+
+                netem
+            })
+            .collect();
+
+        group_netems
+    }
+}
+
+#[derive(Clone)]
+pub struct IntraRegionNetEmConfig {
+    bandwidth_rate_mbps: u64,
+    delay_latency_ms: u64,
+    delay_jitter_ms: u64,
+    delay_correlation_percentage: u64,
+    loss_percentage: u64,
+    loss_correlation_percentage: u64,
+}
+
+impl Default for IntraRegionNetEmConfig {
+    fn default() -> Self {
+        Self {
+            bandwidth_rate_mbps: 10 * 1000, // 10 Gbps
+            delay_latency_ms: 50,
```

Comment:
Curious, how did we come up with these configs and how did we verify they result in a mainnet like setup?

Reply:
Because we couldn't simulate more than 4 inter-regions, Igor and I decided to mandate a minimum latency between validators in the same region. Igor suggested this number based on previewnet experience, but I think this also somewhat falls short of mainnet-like experience. The idea is to tune this in a follow-up to see how close to mainnet we can get as possible.